### ISSUE TITLE
bin/plugin uninstall fails

### ISSUE THREAD

**Issue by joekiller** [elasticsearch/logstash#2689]
From a fresh RPM install which was built from master I run bin/plugin uninstall I get the following generic error. I just use hipchat as an example, this occurs for all the others too.

```
sudo bin/plugin uninstall logstash-output-hipchat
Uninstalling logstash-output-hipchat
Error Bundler::InstallError, retrying 1/10
Error loading RubyGems plugin "/opt/logstash/vendor/bundle/jruby/1.9/gems/jar-dependencies-0.1.7/lib/rubygems_plugin.rb": can't activate jar-dependencies-0.1.2, already activated jar-dependencies-0.1.7 (Gem::LoadError)
Error Bundler::InstallError, retrying 2/10
Error loading RubyGems plugin "/opt/logstash/vendor/bundle/jruby/1.9/gems/jar-dependencies-0.1.7/lib/rubygems_plugin.rb": can't activate jar-dependencies-0.1.2, already activated jar-dependencies-0.1.7 (Gem::LoadError)
Error Bundler::InstallError, retrying 3/10
Error loading RubyGems plugin "/opt/logstash/vendor/bundle/jruby/1.9/gems/jar-dependencies-0.1.7/lib/rubygems_plugin.rb": can't activate jar-dependencies-0.1.2, already activated jar-dependencies-0.1.7 (Gem::LoadError)
Error Bundler::InstallError, retrying 4/10
Error loading RubyGems plugin "/opt/logstash/vendor/bundle/jruby/1.9/gems/jar-dependencies-0.1.7/lib/rubygems_plugin.rb": can't activate jar-dependencies-0.1.2, already activated jar-dependencies-0.1.7 (Gem::LoadError)
Error Bundler::InstallError, retrying 5/10
Error loading RubyGems plugin "/opt/logstash/vendor/bundle/jruby/1.9/gems/jar-dependencies-0.1.7/lib/rubygems_plugin.rb": can't activate jar-dependencies-0.1.2, already activated jar-dependencies-0.1.7 (Gem::LoadError)
Error Bundler::InstallError, retrying 6/10
Error loading RubyGems plugin "/opt/logstash/vendor/bundle/jruby/1.9/gems/jar-dependencies-0.1.7/lib/rubygems_plugin.rb": can't activate jar-dependencies-0.1.2, already activated jar-dependencies-0.1.7 (Gem::LoadError)
Error Bundler::InstallError, retrying 7/10
Error loading RubyGems plugin "/opt/logstash/vendor/bundle/jruby/1.9/gems/jar-dependencies-0.1.7/lib/rubygems_plugin.rb": can't activate jar-dependencies-0.1.2, already activated jar-dependencies-0.1.7 (Gem::LoadError)
Error Bundler::InstallError, retrying 8/10
Error loading RubyGems plugin "/opt/logstash/vendor/bundle/jruby/1.9/gems/jar-dependencies-0.1.7/lib/rubygems_plugin.rb": can't activate jar-dependencies-0.1.2, already activated jar-dependencies-0.1.7 (Gem::LoadError)
Error Bundler::InstallError, retrying 9/10
Error loading RubyGems plugin "/opt/logstash/vendor/bundle/jruby/1.9/gems/jar-dependencies-0.1.7/lib/rubygems_plugin.rb": can't activate jar-dependencies-0.1.2, already activated jar-dependencies-0.1.7 (Gem::LoadError)
Error Bundler::InstallError, retrying 10/10
Error loading RubyGems plugin "/opt/logstash/vendor/bundle/jruby/1.9/gems/jar-dependencies-0.1.7/lib/rubygems_plugin.rb": can't activate jar-dependencies-0.1.2, already activated jar-dependencies-0.1.7 (Gem::LoadError)
Too many retries, aborting, caused by Bundler::InstallError
Uninstall aborted
```

Extremely verbose `DEBUG=true` output: http://pastebin.com/QvLp73Gg

**Comment by colinsurprenant** [comment]
@joekiller thanks, looking at this, trying to reproduce.

**Comment by ph** [comment]
Same issue on issue on macosx

**Comment by ph** [comment]
The bug also occur in a zip artifact, but the command is working inside the logstash directory that we used to generate it.
```
~/e/logstash git:master ❯❯❯ bin/plugin uninstall logstash-output-hipchat                                                                                                    
Uninstalling logstash-output-hipchat
~/e/logstash git:master ❯❯❯
```
(insert @jordansissel captain kirk bundler image here)

When we run the `bin/plugin uninstall` command inside the artifact it will generate a `bundle/config` file in the logstash directory. The file is different from the one generated from a `bin/plugin install`

```diff
diff --git a/uninstall b/.bundle/config
index 0374965..8df439d 100644
--- a/vim
+++ b/.bundle/config
@@ -1,3 +1,4 @@
 ---
-BUNDLE_CLEAN: true
+BUNDLE_PATH: /tmp/logstash-2.0.0.dev/vendor/bundle
+BUNDLE_WITHOUT: development
 BUNDLE_DISABLE_SHARED_GEMS: '
```

**Comment by ph** [comment]
`bundle install` will add the *BUNDLE_PATH* in the .bundler/config but `bundle clean` doesn't. If you look at the bundler clean suite, they will always run a `bundle install` before doing a clean. (https://github.com/bundler/bundler/blob/master/spec/commands/clean_spec.rb)
I'll make a patch that will generate the `.bundle/config` file with our correct option every time.

**Comment by ph** [comment]
@joekiller the https://github.com/elasticsearch/logstash/pull/2691 PR should fix this issue.
Ty again for the awesome bug reports.

**Comment by ph** [comment]
fixed in https://github.com/elasticsearch/logstash/pull/2699

**Comment by ph** [comment]
Should be really fixed in https://github.com/elasticsearch/logstash/pull/2703 we are doing a run of QA.

**Comment by colinsurprenant** [comment]
confirming it is fixed in master by #2703
